Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -159,9 +168,29 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithm::execute(
 
     auto& tracksForSeed = result.value();
     for (auto& track : tracksForSeed) {
-      // Set the seed number, this number decrease by 1 since the seed number
-      // has already been updated
+      auto smoothingResult =
+          Acts::smoothTrack(ctx.geoContext, tracksTemp, track, logger());
+      if (!smoothingResult.ok()) {
+        m_nFailedSmoothing++;
+        ACTS_ERROR("Smoothing for seed "
+                   << iseed << " and track " << track.index()
+                   << " failed with error " << smoothingResult.error());
+        continue;
+      }
+
+      auto extrapolationResult = Acts::extrapolateTrackToReferenceSurface(
+          track, *pSurface, extrapolator, extrapolationOptions,
+          m_cfg.extrapolationStrategy, logger());
+      if (!extrapolationResult.ok()) {
+        m_nFailedExtrapolation++;
+        ACTS_ERROR("Extrapolation for seed "
+                   << iseed << " and track " << track.index()
+                   << " failed with error " << extrapolationResult.error());
+        continue;
```

Comment:
If this happens, we don't copy to the output track container. Is this intended?

Reply:
My thought here was that all output tracks should be expressed at the reference surface. If a track cannot be expressed at the provided reference it will not be of any use to the user.

Alternatively we could flag the track as failed?